packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -2091,19 +2091,42 @@ export default class Meeting extends StatelessWebexPlugin {
    * @memberof Meeting
    */
   getMediaStreams = (mediaDirection, audioVideo = {}, sharePreferences) => {
-    if (mediaDirection && (mediaDirection.sendAudio || mediaDirection.sendVideo || mediaDirection.sendShare)) {
-      return Media.getUserMedia(mediaDirection, audioVideo, sharePreferences)
+    if (
+      mediaDirection &&
+      (
+        mediaDirection.sendAudio ||
+        mediaDirection.sendVideo ||
+        mediaDirection.sendShare
+      )
+    ) {
+      return Media.getSupportedDevice({
+        sendAudio: mediaDirection.sendAudio,
+        sendVideo: mediaDirection.sendVideo
+      })
+        .then(({sendAudio, sendVideo}) =>
+          Media.getUserMedia(
+            {...mediaDirection, sendAudio, sendVideo},
+            audioVideo,
+            sharePreferences
+          ))
         .then((response) => {
           if (!response[0] && !response[1]) {
-            return Promise.reject(new MediaError('Was not able to retrive media Stream'));
+            return Promise.reject(
+              new MediaError('Was not able to retrieve media Stream')
+            );
           }
 
           return Promise.resolve(response);
-        });
+        })
+        .catch((error) => Promise.reject(
```

Comment:
This will also catch the `getUserMedia` call's rejection. Should we put this catch immediately after the `getSupportedDevice` and create another catch for the `getUserMedia` rejection?